pkg/azure/internal/client/error.go

```diff
@@ -0,0 +1,25 @@
+package client
+
+import (
+	"strings"
+
+	"github.com/giantswarm/microerror"
+)
+
+func IsNotFound(err error) bool {
```

Comment:
These have to be duplicated so that there isn't import loop between `azure` and `azure/internal/client`.